File: src/main/java/de/spries/fleetcommander/model/facade/OwnPlayer.java
package de.spries.fleetcommander.model.facade;

import de.spries.fleetcommander.model.core.Player;

public class OwnPlayer extends OtherPlayer {

	public OwnPlayer(Player originalPlayer) {
		super(originalPlayer);
	}

	public Integer getCredits() {
		return originalPlayer.getCredits();
	}
}

File: src/main/kotlin/de/spries/fleetcommander/model/facade/OwnPlayer.kt
package de.spries.fleetcommander.model.facade

import de.spries.fleetcommander.model.core.Player

open class OwnPlayer(originalPlayer: Player) : OtherPlayer(originalPlayer) {

    fun getCredits() = originalPlayer.getCredits()

    fun canAffordFactory() = originalPlayer.canAffordFactory()
}
